### ISSUE TITLE
Right-click on a directory with Windows Explorer in the left panel shows the menus twice

### ISSUE THREAD

**Issue by end2endzone** [end2endzone/ShellAnything#6]

**Comment by end2endzone** [comment]
![image](https://user-images.githubusercontent.com/31753093/57161452-a2d8b800-6db9-11e9-8e3d-2de89800b1cd.png)

**Comment by end2endzone** [comment]
I am unable to identify the cause of the issue. Anyone here can help?

**Comment by end2endzone** [comment]
The issue is still occurring in 0.9.0, branch [feature-issue115](https://github.com/end2endzone/ShellAnything/tree/feature-issue115), 22d3d87b4ec1c733d6990c3d5ed13061a18e0ed4.

**Comment by end2endzone** [comment]
**_Hint_**: Implement a `ScopeLogger` class in file `LoggerHelper.h` which job is to log (or vebose log) when the code enters/leave a scope. This will be much useful when trying to debug this issue.

**Comment by end2endzone** [comment]
I am uncertain why this is happening. Here are two log captures of the same use case : a user right-click on directory C:\Windows\Cursors" :
1. [right-panel.txt](https://github.com/user-attachments/files/16740392/right-panel.txt) made by right-clicking on directory from the **right-panel side** (file list view). 
2. [left-panel.txt](https://github.com/user-attachments/files/16740394/left-panel.txt) made by right-clicking on directory from the **left-panel side** (directory tree view).

(both files were sanitized. See where I inserted `...` for parts that were removed)

----

Investigations notes:
* In left log, 2 instances of `CContextMenu` are created. The second instance is created and destroyed while the first instance is "in use".
* Flags sent to `QueryContextMenu()` in both log files are different.
  1. left-panel: flags=0x00000414=(CMF_EXPLORE|CMF_CANRENAME|CMF_ASYNCVERBSTATE) 
  2. right-panel: flags=0x00020494=(CMF_EXPLORE|CMF_CANRENAME|CMF_ITEMMENU|CMF_ASYNCVERBSTATE)
  3. `CMF_ITEMMENU` is specified for right-panel. CMF_ITEMMENU means :
  0x00000080. The calling application is invoking a shortcut menu on an item in the view (as opposed to the background of the view).
